Lib/pprint.py

```diff
@@ -241,27 +267,28 @@ def _pprint_ordered_dict(self, object, stream, indent, allowance, context, level
             return
         cls = object.__class__
         stream.write(cls.__name__ + '(')
+        recursive_indent = indent + len(cls.__name__) + 1 if not self._block_style else indent
         self._format(list(object.items()), stream,
-                     indent + len(cls.__name__) + 1, allowance + 1,
+                     recursive_indent, allowance + 1,
                      context, level)
         stream.write(')')
 
     _dispatch[_collections.OrderedDict.__repr__] = _pprint_ordered_dict
 
     def _pprint_list(self, object, stream, indent, allowance, context, level):
-        stream.write('[')
+        stream.write(self._format_block_start('[', indent))
         self._format_items(object, stream, indent, allowance + 1,
                            context, level)
-        stream.write(']')
+        stream.write(self._format_block_end(']', indent))
```

Comment:
You have a lot of repeated code here conditional on `self._block_style`. How about defining a context manager to clean this up?

```python
with self._block(stream, indent, '[', ']') as new_indent:
    self._format_items(object, stream, new_indent, allowance + 1,
                       context, level)
```